src/app/api/schemas.py

```diff
@@ -153,10 +153,15 @@ class EventOut(EventIn, _CreatedAt, _Id):
 
 
 # Device
+class SoftwareHash(BaseModel):
```

Comment:
Quick question: how is the software hash computed ? (hash of the commit yes,  but of which repo ?) We have pyro-engine but also pyro-vision no ? How will the device be able to know which repo it needs to pull ?

Reply:
The hash is computed on device, so that the convention about how the deterministic function is computed is deferred to your choice :)
The comparison will be performed on the device anyway

Reply:
Mhmm okay. Not sure yet how the hash will be computed, especially if we have several repo to update on the rpi. Don't know yet what will be installed on the rpi and what might need update. 

If there are several repo, how will the rpi how which repo to update ? If we have a unique hash (hash of hashes let's say), the rpi will update everything (pull all the repos using git or we can directly DL the source code as tar.gz for each repo).
Another way could be to add a parameter to the route to get the rpi to check the hash for every repo which might need updating.

We will need to discuss this @pyronear/hardware. Depending on our use case, I think that DL the source code might be easier (for instance we won't need to install and use git on the rpi) but then the .env file would need to be in another directory (not to get erase when DL the source code during the update).

However it is maybe too soon to talk about it and decide if we still don't know what form will take the tools that will be installed on the rpi (do we know to have the source code, will we use a Docker image etc etc ?) 

Anyways this is not the right place to discuss this. I will open an issue on the appropriate repo asap.

Reply:
Well that's the point: the computation should not be decided by the API :ok_hand: 
If there are several repos, it's rather easy : computing the hash of the concatenation of all git commit hashes.

But either way, in my opinion, as you said, this will not be handled on the API side